be/src/exec/workgroup/pipeline_executors_manager.cpp

```diff
@@ -0,0 +1,207 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exec/workgroup/pipeline_executors_manager.h"
+
+#include "work_group.h"
+
+namespace starrocks::workgroup {
+
+ExecutorsManager::ExecutorsManager(WorkGroupManager* parent, PipelineExecutorsConfig conf)
+        : _parent(parent), _conf(std::move(conf)) {
+    _wg_to_cpuids[COMMON_WORKGROUP] = _conf.total_cpuids;
+    for (auto cpuid : _conf.total_cpuids) {
+        _cpu_owners[cpuid].set_wg(COMMON_WORKGROUP);
+    }
+}
+
+void ExecutorsManager::close() const {
+    for_each_executors([](auto& executors) { executors.close(); });
+}
+
+Status ExecutorsManager::start_shared_executors() {
+    _shared_executors =
+            std::make_unique<PipelineExecutors>(_conf, "com", _conf.total_cpuids, std::vector<CpuUtil::CpuIds>{});
```

Comment:
PipelineExecutors is set of executors working for the resource group, can we use another name which indicate it is a set.  for an example:
1. PipelineExecutorSet,
2. PipelineExecutorAssembly,
3. PipelineExecutorFederal,
4. PipelineExecutorChamber,
5. PipelineExecutorCrew

Reply:
Good suggestions. Renamed to `PipelineExecutorSet`.